src/java/util/ImageUtils.java

```diff
@@ -45,4 +49,70 @@ public static BufferedImage scaleImage(BufferedImage image, int width, int heigh
         return imageScaled;
     }
 
+    public static void writePNGFromMasks(FloatMask redMask, FloatMask greenMask, FloatMask blueMask, float scaleMultiplier, String path) throws IOException {
+        int size = redMask.getSize();
+        if (size != greenMask.getSize() || size != blueMask.getSize()) {
+            throw new IllegalArgumentException("Masks not the same size: redMask is " + redMask.getSize() +
+                    ", greenMask is " + greenMask.getSize() + " and blueMask is " + blueMask.getSize());
+        }
+        int length = size * size * 3;
+        int index = -1;
+        final byte[] byteArray = new byte[length];
+        for (int x = 0; x < size; x++) {
+            for (int y = 0; y < size; y++) {
+                byte redByte = (byte) ((Float) (redMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                byte greenByte = (byte) ((Float) (greenMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                byte blueByte = (byte) ((Float) (blueMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                index += 3;
+                byteArray[index - 2] = redByte;
+                byteArray[index - 1] = greenByte;
+                byteArray[index] = blueByte;
```

Comment:
you can increment at the end of the loop to allow it to start at 0. Additionally you could compute the index from x and y as (x * size + y) * 3